src/Catalyst.Core.Modules.Web3/Web3EthApiExtensions.cs

```diff
@@ -0,0 +1,183 @@
+using Catalyst.Protocol.Transaction;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using LibP2P;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Dirichlet.Numerics;
+using Nethermind.Evm.Tracing;
+
+namespace Catalyst.Core.Modules.Web3 
+{
+    public static class Web3EthApiExtensions
+    {
+        public static DeltaWithCid GetLatestDeltaWithCid(this IWeb3EthApi api)
+        {
+            return GetDeltaWithCid(api, api.DeltaResolver.LatestDelta);
+        }
+
+        public static DeltaWithCid GetDeltaWithCid(this IWeb3EthApi api, Cid cid)
+        {
+            if (!api.DeltaCache.TryGetOrAddConfirmedDelta(cid, out Delta delta))
+            {
+                throw new Exception($"Delta not found '{cid}'");
+            }
+
+            return new DeltaWithCid
+            {
+                Delta = delta,
+                Cid = cid
+            };
+        }
+
+        public static bool TryGetDeltaWithCid(this IWeb3EthApi api, BlockParameter block, out DeltaWithCid delta)
+        {
+            Cid cid;
+            switch (block.Type)
+            {
+                case BlockParameterType.Earliest:
+                    cid = api.DeltaCache.GenesisHash;
+                    break;
+                case BlockParameterType.Latest:
+                    cid = api.DeltaResolver.LatestDelta;
+                    break;
+                case BlockParameterType.Pending:
+                    cid = api.DeltaResolver.LatestDelta;
+                    break;
+                case BlockParameterType.BlockNumber:
+                    var blockNumber = block.BlockNumber.Value;
+                    if (!api.DeltaResolver.TryResolve(blockNumber, out cid))
+                    {
+                        delta = default;
+                        return false;
+                    }
+
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            delta = api.GetDeltaWithCid(cid);
+            return true;
+        }
+
+        public static PublicEntry ToPublicEntry(this IWeb3EthApi api, TransactionForRpc transactionCall, Keccak root)
+        {
+            return new PublicEntry
+            {
+                Nonce = (ulong) api.StateReader.GetNonce(root, transactionCall.From),
+                SenderAddress = transactionCall.From.Bytes.ToByteString(),
+                ReceiverAddress = transactionCall.To?.Bytes.ToByteString() ?? ByteString.Empty,
+                GasLimit = (ulong) transactionCall.Gas.GetValueOrDefault(),
+                GasPrice = transactionCall.GasPrice.GetValueOrDefault().ToUint256ByteString(),
+                Amount = transactionCall.Value.GetValueOrDefault().ToUint256ByteString(), 
+                Data = transactionCall.Data?.ToByteString() ?? ByteString.Empty,
+                Timestamp = Timestamp.FromDateTime(DateTime.UtcNow) //TODO: this should be set by user and signed
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)